workers/unity/Assets/Tests/EditmodeTests/Correctness/Subscriptions/RequireablesReenableTests.cs

```diff
@@ -0,0 +1,117 @@
+                {
+                    Assert.IsNull(behaviour.CommandReceiver);
+                    Assert.IsNull(behaviour.CommandSender);
+                })
+                .Step(world =>
+                    world.Connection.ChangeAuthority(EntityId, TestCommands.ComponentId, Authority.Authoritative))
+                .Step((world, behaviour) =>
+                {
+                    Assert.IsNotNull(behaviour.CommandReceiver);
+                    Assert.IsNotNull(behaviour.CommandSender);
+                    Assert.IsTrue(behaviour.CommandReceiver.IsValid);
+                    Assert.IsTrue(behaviour.CommandSender.IsValid);
+                });
+        }
+
+        [Test]
+        public void Readers_and_Writers_are_valid_after_reinjection()
+        {
+            World
+                .Step(world => world.Connection.CreateEntity(EntityId, GetTemplate()))
+                .Step(world =>
+                    world.Connection.ChangeAuthority(EntityId, TestCommands.ComponentId, Authority.Authoritative))
+                .Step(world =>
+                {
+                    var (_, behaviour) = world.CreateGameObject<ReaderTestBehaviour>(EntityId);
+                    return behaviour;
+                })
+                .Step((world, behaviour) =>
+                {
+                    Assert.IsNotNull(behaviour.Writer);
+                    Assert.IsNotNull(behaviour.Reader);
+                    Assert.IsTrue(behaviour.Writer.IsValid);
+                    Assert.IsTrue(behaviour.Reader.IsValid);
+                })
+                .Step(world =>
+                    world.Connection.ChangeAuthority(EntityId, TestCommands.ComponentId, Authority.NotAuthoritative))
+                .Step((world, behaviour) =>
+                {
+                    Assert.IsNull(behaviour.Writer);
+                    Assert.IsNull(behaviour.Reader);
+                })
+                .Step(world =>
+                    world.Connection.ChangeAuthority(EntityId, TestCommands.ComponentId, Authority.Authoritative))
+                .Step((world, behaviour) =>
+                {
+                    Assert.IsNotNull(behaviour.Writer);
+                    Assert.IsNotNull(behaviour.Reader);
+                    Assert.IsTrue(behaviour.Writer.IsValid);
+                    Assert.IsTrue(behaviour.Reader.IsValid);
+                });
+        }
+
+        private static EntityTemplate GetTemplate()
+        {
+            var template = new EntityTemplate();
+            template.AddComponent(new Position.Snapshot(), "worker");
+            template.AddComponent(new TestCommands.Snapshot(), "worker");
+            return template;
+        }
+
+        private class CommandSenderTestBehaviour : MonoBehaviour
+        {
+            [Require] public TestCommandsCommandReceiver CommandReceiver;
+            [Require] public PlayerHeartbeatClientCommandSender CommandSender;
+        }
+
+        private class ReaderTestBehaviour : MonoBehaviour
```

Comment:
I got a bit confused by the name of this, perhaps `ReaderWriterTestBehaviour` and similarly `CommandSenderReceiverTestBehaviour`?

Reply:
Sure thing